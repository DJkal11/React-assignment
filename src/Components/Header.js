

function Header() {
  return (
    
<nav data-testid="header" className="bg-white border-b-2 shadow-sm border-gray-100 px-2 sm:px-4 py-4  ">
  <div className="container flex flex-wrap justify-between items-center mx-auto">  
        <span className="text-orange-500 text-2xl font-semibold">Dashboard</span>
  </div>
</nav>

  );
}

export default Header;