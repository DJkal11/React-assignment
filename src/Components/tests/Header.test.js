import { render, screen, cleanup } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";

import Header from "../Header"

afterEach(() => {
  cleanup()
})

test("should render header component", () => {
    render(<Header/>)
    const headerComponent = screen.getByTestId('header');
    expect(headerComponent).toBeInTheDocument();
})

test("Header includes correct html innerText", () => {
    render(<Header/>)
    const headerComponent = screen.getByTestId('header');
    expect(headerComponent).toHaveTextContent("Dashboard");
})


test("renders Header snapshot", () => {
    const newHeader = renderer.create(<Header/>).toJSON();
    expect(newHeader).toMatchSnapshot();
})