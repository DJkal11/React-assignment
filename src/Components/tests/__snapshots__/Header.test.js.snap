// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header snapshot 1`] = `
<nav
  className="bg-white border-b-2 shadow-sm border-gray-100 px-2 sm:px-4 py-4  "
  data-testid="header"
>
  <div
    className="container flex flex-wrap justify-between items-center mx-auto"
  >
    <span
      className="text-orange-500 text-2xl font-semibold"
    >
      Dashboard
    </span>
  </div>
</nav>
`;
